src/libsystemd/sd-journal/test-compress.c

```diff
@@ -204,6 +205,7 @@ _unused_ static void test_compress_stream(const char *compression,
         assert_se((dst2 = mkostemp_safe(pattern2)) >= 0);
 
         assert_se(stat(srcfile, &st) == 0);
+        assert_se((size_t)st.st_size == uncompressed_size);
```

Comment:
you cast the likely large type to a smaller type here before comparing, thus on 32bit systems this might theoretically break. please cast up, not down, i.e. `(size_t)` → `(uint64_t)`